indexing-hadoop/src/main/java/org/apache/druid/indexer/hadoop/DatasourceRecordReader.java

```diff
@@ -163,4 +189,157 @@ public void close() throws IOException
       FileUtils.deleteDirectory(dir);
     }
   }
+
+  public static class SegmentReader implements Closeable
```

Comment:
I think it would be easier to review this if we leave this class in a separate file and just rename `IngestSegmentFirehose` to `SegmentReader`.

Reply:
left this as an internal class btw because its only used internally to `DatasourceRecordReader`